src/Neo/SmartContract/Native/TokenManagement.cs

```diff
@@ -0,0 +1,190 @@
+        if (engine.SnapshotCache.Contains(key))
+            throw new InvalidOperationException($"{name} already exists.");
+        var state = new TokenState
+        {
+            Owner = owner,
+            Name = name,
+            Symbol = symbol,
+            Decimals = decimals,
+            TotalSupply = BigInteger.Zero
+        };
+        engine.SnapshotCache.Add(key, new(state));
+        Notify(engine, "Created", tokenid);
+        return tokenid;
+    }
+
+    [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates)]
+    public TokenState? GetTokenInfo(ApplicationEngine engine, UInt160 tokenid)
+    {
+        StorageKey key = CreateStorageKey(Prefix_TokenState, tokenid);
+        return engine.SnapshotCache.TryGet(key)?.GetInteroperable<TokenState>();
+    }
+
+    [ContractMethod(CpuFee = 1 << 17, StorageFee = 1 << 7, RequiredCallFlags = CallFlags.All)]
+    internal async Task Mint(ApplicationEngine engine, UInt160 tokenid, UInt160 account, BigInteger amount)
+    {
+        ArgumentOutOfRangeException.ThrowIfNegativeOrZero(amount);
+        ArgumentOutOfRangeException.ThrowIfGreaterThan(amount, MaxMintAmount);
+        AddTotalSupply(engine, tokenid, amount, assertOwner: true);
+        AddBalance(engine.SnapshotCache, tokenid, account, amount);
+        await PostTransferAsync(engine, tokenid, null, account, amount, StackItem.Null, callOnPayment: true);
+    }
+
+    [ContractMethod(CpuFee = 1 << 17, RequiredCallFlags = CallFlags.All)]
+    internal async Task Burn(ApplicationEngine engine, UInt160 tokenid, UInt160 account, BigInteger amount)
+    {
+        ArgumentOutOfRangeException.ThrowIfNegativeOrZero(amount);
+        ArgumentOutOfRangeException.ThrowIfGreaterThan(amount, MaxMintAmount);
+        AddTotalSupply(engine, tokenid, -amount, assertOwner: true);
+        if (!AddBalance(engine.SnapshotCache, tokenid, account, -amount))
+            throw new InvalidOperationException("Insufficient balance to burn.");
+        await PostTransferAsync(engine, tokenid, account, null, amount, StackItem.Null, callOnPayment: false);
+    }
+
+    [ContractMethod(CpuFee = 1 << 17, StorageFee = 1 << 7, RequiredCallFlags = CallFlags.All)]
+    internal async Task<bool> Transfer(ApplicationEngine engine, UInt160 tokenid, UInt160 from, UInt160 to, BigInteger amount, StackItem data)
+    {
+        ArgumentOutOfRangeException.ThrowIfNegative(amount);
+        StorageKey key = CreateStorageKey(Prefix_TokenState, tokenid);
+        TokenState token = engine.SnapshotCache.TryGet(key)?.GetInteroperable<TokenState>()
+            ?? throw new InvalidOperationException("The token id does not exist.");
+        if (!engine.CheckWitnessInternal(from)) return false;
+        if (!amount.IsZero && from != to)
+        {
+            if (!AddBalance(engine.SnapshotCache, tokenid, from, -amount))
+                return false;
+            AddBalance(engine.SnapshotCache, tokenid, to, amount);
```

Comment:
```suggestion
            if (!AddBalance(engine.SnapshotCache, tokenid, to, amount)) throw new InvalidOperationException("Balance can't be increased");
```

Reply:
If the `amount` is positive, then `AddBalance` cannot return `false`.